examples/sparse/c_and_s.py

```diff
@@ -0,0 +1,143 @@
+        super().__init__()
+
+        self.num_layers = num_layers
+        self.alpha = alpha
+
+    ###########################################################################
+    # (HIGHLIGHT) Compute Label Propagation with Sparse Matrix API
+    ###########################################################################
+    @torch.no_grad()
+    def forward(self, A_hat, label):
+        Y = label
+        for _ in range(self.num_layers):
+            Y = self.alpha * A_hat @ Y + (1 - self.alpha) * label
+            Y = Y.clamp_(0.0, 1.0)
+        return Y
+
+
+class CorrectAndSmooth(nn.Module):
+    def __init__(self):
+        super().__init__()
+
+        self.lp = LabelPropagation()
+
+    def correct(self, A_hat, label, soft_label, mask):
+        # Compute error.
+        error = torch.zeros_like(soft_label)
+        error[mask] = label[mask] - soft_label[mask]
+
+        # Smooth error.
+        smoothed_error = self.lp(A_hat, error)
+
+        # Autoscale.
+        sigma = error[mask].abs()
+        sigma = sigma.sum() / sigma.shape[0]
+        scale = sigma / smoothed_error.abs().sum(dim=1, keepdim=True)
+        scale[scale.isinf() | (scale > 1000)] = 1.0
+
+        # Correct.
+        result = soft_label + scale * smoothed_error
+        return result
+
+    def smooth(self, A_hat, label, soft_label, mask):
+        soft_label[mask] = label[mask].float()
+        return self.lp(A_hat, soft_label)
+
+
+def evaluate(g, pred):
+    label = g.ndata["label"]
+    val_mask = g.ndata["val_mask"]
+    test_mask = g.ndata["test_mask"]
+
+    # Compute accuracy on validation/test set.
+    val_acc = (pred[val_mask] == label[val_mask]).float().mean()
+    test_acc = (pred[test_mask] == label[test_mask]).float().mean()
+    return val_acc, test_acc
+
+
+def train(base_model, model, g, A_hat, X):
+    label = g.ndata["label"]
+    train_mask = g.ndata["train_mask"]
+
+    # Stage1: Train the base model.
+    optimizer = Adam(base_model.parameters(), lr=0.01)
+
+    for epoch in range(10):
+        # Forward.
+        base_model.train()
+        logits = base_model(X)
+
+        # Compute loss with nodes in training set.
+        loss = F.cross_entropy(logits[train_mask], label[train_mask])
+
+        # Backward.
+        optimizer.zero_grad()
+        loss.backward()
+        optimizer.step()
+
+        # Compute prediction.
+        base_model.eval()
+        logits = base_model(X)
+        pred = logits.argmax(dim=1)
+
+        # Evaluate the prediction.
+        val_acc, test_acc = evaluate(g, pred)
+        print(
+            f"Base model, In epoch {epoch}, loss: {loss:.3f}, "
+            f"val acc: {val_acc:.3f}, test acc: {test_acc:.3f}"
+        )
+
+    # Stage2: Correct and Smooth.
+    soft_label = F.softmax(logits, dim=1)
+    label = F.one_hot(label)
+    soft_label = model.correct(A_hat, label, soft_label, train_mask)
+    soft_label = model.smooth(A_hat, label, soft_label, train_mask)
```

Comment:
Let's write them as plain functions:
```
soft_label = correct(A_hat, label, soft_label, train_mask)
soft_label = smooth(A_hat, label, soft_label, train_mask)
```

In this way, you don't need to pass the `model` object anymore and you can rename `base_model` to `model`.